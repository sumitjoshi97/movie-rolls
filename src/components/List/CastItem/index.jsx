import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class CastItem extends PureComponent {
  static propTypes = {
    poster: PropTypes.string,
    name: PropTypes.string.isRequired,
    role: PropTypes.string
  }

  render() {
    const { poster, name, role } = this.props

    return (
      <div className='list-item'>
        <img
          src={`https://image.tmdb.org/t/p/w185/${poster}`}
          alt={name}
          className='list-item__img'
        />
        <div className='list__item__name'>{name}</div>
        <div className='list__item__name__sub'>{role}</div>
      </div>
    )
  }
}
