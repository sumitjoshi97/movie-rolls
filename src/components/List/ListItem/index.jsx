import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import * as actions from '../../../actions'

class ListItem extends PureComponent {
  static propTypes = {
    pageId: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    poster: PropTypes.string,
    name: PropTypes.string.isRequired,
    setType: PropTypes.func.isRequired
  }

  render() {
    const { pageId, type, poster, name, setType } = this.props

    return (
      <div className='list-item'>
        <Link to={`/details/${pageId}`} onClick={() => setType(type)}>
          <img
            src={`https://image.tmdb.org/t/p/w185/${poster}`}
            alt={name}
            className='list-item__img'
          />
          <div className='list__item__name'>{name}</div>
        </Link>
        )
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  setType: type => dispatch(actions.setItemType(type))
})

export default connect(
  null,
  mapDispatchToProps
)(ListItem)
