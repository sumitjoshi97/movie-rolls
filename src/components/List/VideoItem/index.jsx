import React from 'react'
import PropTypes from 'prop-types'

const VideoItem = ({ video, name, category }) => (
  <div className='list-item'>
    <iframe
      className='list-item__video'
      title='1'
      width='280'
      src={`https://www.youtube.com/embed/${video}`}
    />
    <div className='list-item__name'>{name}</div>
    <div className='list-item__name__sub'>{category}</div>
  </div>
)

VideoItem.propTypes = {
  video: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  category: PropTypes.string
}

export default VideoItem
