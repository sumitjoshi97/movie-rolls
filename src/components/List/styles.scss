.list {
  padding: 6rem 6rem 0 6rem;
  background-color: #1a1a1a;
  &__header {
    text-transform: capitalize;
    padding-bottom: 1rem;
  }
}

.list-item {
  display: inline-block;
  padding: 2rem;
  &__img {
    height: 18rem;
    background: #000;
    width: 14rem;
  }

  &__video {
    display: inline-block;
    height: 18rem;
    outline: none;
    border: none;
  }

  &__name {
  }
}
