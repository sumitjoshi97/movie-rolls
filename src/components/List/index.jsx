import React, { PureComponent } from 'react'
import Slider from 'react-slick'

import ListItem from './ListItem'
import CastItem from './CastItem'
import VideoItem from './VideoItem'
import NextArrow from '../Arrows/Next'
import PrevArrow from '../Arrows/Prev'

import './styles.scss'

export default class List extends PureComponent {
  renderList = () => {
    const { type, items } = this.props

    switch (type) {
      case 'cast':
        return items.map(item => (
          <CastItem
            key={item.id}
            name={item.name}
            poster={item.profile_path}
            role={item.character}
          />
        ))

      case 'video':
        return items.map(item => (
          <VideoItem
            key={item.id}
            name={item.name}
            video={item.key}
            category={item.type}
          />
        ))

      //for movies / tv shows
      default:
        return items.map(item => (
          <ListItem
            key={item.id}
            type={type}
            pageId={item.id}
            name={item.title}
            rating={item.vote_average}
            poster={item.poster_path}
          />
        ))
    }
  }

  render() {
    const sliderOptions = {
      arrows: true,
      dots: false,
      speed: 400,
      slidesToShow: this.props.type === 'video' ? 4 : 7,
      swipeToSlide: true,
      infinite: false,
      nextArrow: <NextArrow />,
      prevArrow: <PrevArrow />
    }
    const { name } = this.props
    return (
      <div className='list'>
        <h2 className='list__header'>{name}</h2>
        <Slider {...sliderOptions}>{this.renderList()}</Slider>
      </div>
    )
  }
}
