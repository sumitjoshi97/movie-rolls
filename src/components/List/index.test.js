import React from 'react'
import { shallow } from 'enzyme'

import List from './index'

describe('list unit tests', () => {
  const type = 'movie'
  const items = [
    {
      id: '01',
      title: 'test',
      poster_path: 'test_path.png'
    }
  ]
  const name = 'test title'

  const props = {
    type,
    items,
    name
  }

  const listWrapper = shallow(<List {...props} />)
  test('should render properly', () => {
    expect(listWrapper).toMatchSnapshot()
  })

  test('should have header of title - `test title`', () => {
    expect(listWrapper.find('.list__header').text()).toEqual(name)
  })

  test('should render a ListItem component of type `movie`', () => {
    expect(listWrapper.find({ type: 'movie' })).toHaveLength(1)
  })

  test('should render a ListItem component of type `tv`', () => {
    const props = {
      type: 'tv',
      items,
      name
    }
    const listWrapper = shallow(<List {...props} />)
    expect(listWrapper.find({ type: 'tv' })).toHaveLength(1)
  })

  test('should render a ListItem component of type `video`', () => {
    const props = {
      type: 'video',
      items: [
        {
          id: '01',
          name: 'test video',
          key: 'video',
          category: 'trailer'
        }
      ],
      name
    }
    const listWrapper = shallow(<List {...props} />)
    expect(listWrapper.find({ name: 'test video' })).toHaveLength(1)
  })

  test('should render a ListItem component of type `cast`', () => {
    const props = {
      type: 'cast',
      items: [
        {
          id: '01',
          name: 'test cast',
          poster: 'test_path.png',
          role: 'test role'
        }
      ],
      name
    }
    const listWrapper = shallow(<List {...props} />)
    expect(listWrapper.find({ name: 'test cast' })).toHaveLength(1)
  })
})
