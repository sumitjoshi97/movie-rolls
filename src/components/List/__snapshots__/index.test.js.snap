// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list unit tests should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    items={
      Array [
        Object {
          "id": "01",
          "poster_path": "test_path.png",
          "title": "test",
        },
      ]
    }
    name="test title"
    type="movie"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="list__header"
        >
          test title
        </h2>,
        <Slider
          arrows={true}
          dots={false}
          infinite={false}
          nextArrow={<NextArrow />}
          prevArrow={<PrevArrow />}
          responsive={
            Array [
              Object {
                "breakpoint": 1440,
                "settings": Object {
                  "slidesToShow": 6,
                },
              },
              Object {
                "breakpoint": 1200,
                "settings": Object {
                  "slidesToShow": 5,
                },
              },
              Object {
                "breakpoint": 1000,
                "settings": Object {
                  "slidesToShow": 5,
                },
              },
              Object {
                "breakpoint": 680,
                "settings": Object {
                  "slidesToShow": 4,
                },
              },
              Object {
                "breakpoint": 570,
                "settings": Object {
                  "slidesToShow": 3,
                },
              },
              Object {
                "breakpoint": 480,
                "settings": Object {
                  "slidesToShow": 2,
                },
              },
            ]
          }
          slidesToShow={7}
          speed={400}
          swipeToSlide={true}
        >
          <Connect(ListItem)
            name="test"
            pageId="01"
            poster="test_path.png"
            type="movie"
          />
        </Slider>,
      ],
      "className": "list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test title",
          "className": "list__header",
        },
        "ref": null,
        "rendered": "test title",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrows": true,
          "children": Array [
            <Connect(ListItem)
              name="test"
              pageId="01"
              poster="test_path.png"
              type="movie"
            />,
          ],
          "dots": false,
          "infinite": false,
          "nextArrow": <NextArrow />,
          "prevArrow": <PrevArrow />,
          "responsive": Array [
            Object {
              "breakpoint": 1440,
              "settings": Object {
                "slidesToShow": 6,
              },
            },
            Object {
              "breakpoint": 1200,
              "settings": Object {
                "slidesToShow": 5,
              },
            },
            Object {
              "breakpoint": 1000,
              "settings": Object {
                "slidesToShow": 5,
              },
            },
            Object {
              "breakpoint": 680,
              "settings": Object {
                "slidesToShow": 4,
              },
            },
            Object {
              "breakpoint": 570,
              "settings": Object {
                "slidesToShow": 3,
              },
            },
            Object {
              "breakpoint": 480,
              "settings": Object {
                "slidesToShow": 2,
              },
            },
          ],
          "slidesToShow": 7,
          "speed": 400,
          "swipeToSlide": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "movie01",
            "nodeType": "class",
            "props": Object {
              "name": "test",
              "pageId": "01",
              "poster": "test_path.png",
              "type": "movie",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="list__header"
          >
            test title
          </h2>,
          <Slider
            arrows={true}
            dots={false}
            infinite={false}
            nextArrow={<NextArrow />}
            prevArrow={<PrevArrow />}
            responsive={
              Array [
                Object {
                  "breakpoint": 1440,
                  "settings": Object {
                    "slidesToShow": 6,
                  },
                },
                Object {
                  "breakpoint": 1200,
                  "settings": Object {
                    "slidesToShow": 5,
                  },
                },
                Object {
                  "breakpoint": 1000,
                  "settings": Object {
                    "slidesToShow": 5,
                  },
                },
                Object {
                  "breakpoint": 680,
                  "settings": Object {
                    "slidesToShow": 4,
                  },
                },
                Object {
                  "breakpoint": 570,
                  "settings": Object {
                    "slidesToShow": 3,
                  },
                },
                Object {
                  "breakpoint": 480,
                  "settings": Object {
                    "slidesToShow": 2,
                  },
                },
              ]
            }
            slidesToShow={7}
            speed={400}
            swipeToSlide={true}
          >
            <Connect(ListItem)
              name="test"
              pageId="01"
              poster="test_path.png"
              type="movie"
            />
          </Slider>,
        ],
        "className": "list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test title",
            "className": "list__header",
          },
          "ref": null,
          "rendered": "test title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrows": true,
            "children": Array [
              <Connect(ListItem)
                name="test"
                pageId="01"
                poster="test_path.png"
                type="movie"
              />,
            ],
            "dots": false,
            "infinite": false,
            "nextArrow": <NextArrow />,
            "prevArrow": <PrevArrow />,
            "responsive": Array [
              Object {
                "breakpoint": 1440,
                "settings": Object {
                  "slidesToShow": 6,
                },
              },
              Object {
                "breakpoint": 1200,
                "settings": Object {
                  "slidesToShow": 5,
                },
              },
              Object {
                "breakpoint": 1000,
                "settings": Object {
                  "slidesToShow": 5,
                },
              },
              Object {
                "breakpoint": 680,
                "settings": Object {
                  "slidesToShow": 4,
                },
              },
              Object {
                "breakpoint": 570,
                "settings": Object {
                  "slidesToShow": 3,
                },
              },
              Object {
                "breakpoint": 480,
                "settings": Object {
                  "slidesToShow": 2,
                },
              },
            ],
            "slidesToShow": 7,
            "speed": 400,
            "swipeToSlide": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "movie01",
              "nodeType": "class",
              "props": Object {
                "name": "test",
                "pageId": "01",
                "poster": "test_path.png",
                "type": "movie",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
